### PR TITLE
Custom action tests for GCP Datadog Bitbucket

### PR DESCRIPTION
**Pull request by YakirOren**

### CODE REVIEW

File: plugins/bitbucket/custom_actions_test.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	router := gin.New()
+
+	v1 := router.Group("/1.0")
+	v1.PUT("/groups/:workspaceID/:groupSlug/members/:user", func(c *gin.Context) {
+		group := map[string][]string{
+			"blink/blink": {"Member1"},
+		}
+
+		workspaceID := c.Param("workspaceID")
+		// log.Println(workspaceID)
```
Comment (ozhey) [inline plugins/bitbucket/custom_actions_test.go:115]: Forgot to remove?

File: plugins/datadog/custom_actions_test.go [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+		},
+		{
+			name: "",
+			args: args{
+				fields: "sldfjlsd",
+			},
+			want:    nil,
+			wantErr: errors.New("failed to unmarshal 'fields' param"),
+		},
+	}
```
Comment (ozhey) [inline plugins/datadog/custom_actions_test.go:99]: no test names?

File: plugins/datadog/custom_actions_test.go [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+			wantErr: errors.New("failed to convert 'customer impacted' to boolean"),
+		},
+		{
+			name: "",
+			args: args{
+				fields: "sldfjlsd",
+			},
+			want:    nil,
+			wantErr: errors.New("failed to unmarshal 'fields' param"),
+		},
```
Comment (ozhey) [inline plugins/datadog/custom_actions_test.go:174]: If you don't want to use the test name it's fine, but then just remove it

File: plugins/datadog/custom_actions_test.go [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+		messageParam    string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []byte
+		wantErr error
+	}{
+		{
+			name: "",
```
Comment (ozhey) [inline plugins/datadog/custom_actions_test.go:215]: same heere

File: plugins/datadog/custom_actions_test.go [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+				incidentID: "1",
+				link:       "fsadfasd",
+				title:      "asdfsadf",
+			},
+			want:    []byte("{}"),
+			wantErr: nil,
+		},
+
+		{
+			name: "",
```
Comment (ozhey) [inline plugins/datadog/custom_actions_test.go:264]: same

File: plugins/datadog/custom_actions_test.go [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+		title      string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []byte
+		wantErr error
+	}{
+		{
+			name: "",
```
Comment (ozhey) [inline plugins/datadog/custom_actions_test.go:253]: same

File: plugins/plugins.go [hunk 7]
```diff
@@ -34,13 +34,13 @@ var Plugins = map[string]types.Plugin{
...
 	"azure-devops":      azure_devops.GetNewAzureDevopsPlugin(),
-	"bitbucket":         bitbucket.GetNewBitbucketPlugin(),
+	"bitbucket":         bitbucket.GetNewBitbucketPlugin("https://api.bitbucket.org/"),
 	"crowdstrike":       crowdstrike.GetNewCrowdStrikePlugin(),
 	"datadog":           datadog.GetNewDatadogPlugin(),
 	"discord":           discord.GetNewDiscordPlugin(),
 	"elasticsearch":     elasticsearch.GetNewElasticSearchPlugin(),
 	"ermetic":           ermetic.GetNewErmeticPlugin(),
-	"gcp":               gcp.GetNewGcpPlugin(),
+	"gcp":               gcp.GetNewGcpPlugin("https://admin.googleapis.com/"),
```
Comment (ozhey) [inline plugins/plugins.go:43]: I don't think all endpoints go to to the admin subdomain. @shirleygordon-blink 
Should be googleapis.com like before

**Review by ozhey — CHANGES_REQUESTED** [review]

File: plugins/plugins.go (hunk 7 above)
Comment (YakirOren) [reply plugins/plugins.go:43]: The custom actions are for google workspace, which goes to the admin subdomain.

Comment (ozhey) [reply plugins/plugins.go:43]: The default request url is being enforced on all actions using gcp connection, not only custom actions

**Comment by ozhey** [comment]
you need to merge from main in order for the tests to pass

**Review by ozhey — APPROVED** [review]

**Pull request merged**